components/Starknet/modules/architecture-and-concepts/pages/smart-contracts/compiled_class_hash.adoc

```diff
@@ -0,0 +1,39 @@
+[id="compiled_class_hash"]
+= Compiled class hash
+
+
+The compiled class hash is a cryptographic hash that results from the compilation process of a Cairo class
+
+from its intermediate representation (Sierra) to Cairo assembly (Casm). This process is managed by the Sierra→Casm compiler.
+
+The compiled class hash is a crucial component within the Starknet ecosystem, specifically used to uniquely identify the output generated by the Sierra to Casm (Cairo assembly) compiler. This hash ensures that each compiled class output is distinct and verifiable.
+
+Within Starknet, Cairo classes forming part of the state commitment are specified with Sierra, acting as an intermediate representation bridging Cairo and Cairo assembly (Casm). Despite this, provers operate solely with Casm.
+
+To circumvent the necessity of recompiling from Sierra to Casm each block in which the class is deployed, the state commitment necessitates relevant information concerning the corresponding Cairo assembly. This requisite information is encapsulated within the compiled class hash.
```

Comment:
```suggestion
In order to avoid recompiling, from Sierra to Casm, each block in which the class is deployed, the state commitment gets the information it needs about the corresponding Casm from the the information contained in the compiled class hash.
```